docs/source/developer/accessibility.rst

```diff
@@ -0,0 +1,172 @@
+<https://jupyter-accessibility.readthedocs.io/en/latest/resources/map-jupyterlab-frontend-architecture/README.html>`__?
+
+Best practices while developing
+-------------------------------
+
+JupyterLab is a web application, therefore the following two standards apply:
+
+- WCAG - `Web Content Accessibility Guidelines
+  <https://www.w3.org/WAI/standards-guidelines/wcag/>`__
+- ARIA - `Accessible Rich Internet Applications
+  <https://www.w3.org/WAI/standards-guidelines/aria/>`__
+
+These are good places to familiarize yourself with accessibility best practices
+for developing web sites (WCAG) and web applications (ARIA). Note that although
+WCAG was created primarily for static web sites, the guidelines are nonetheless
+applicable to web apps like JupyterLab.
+
+One resource that is often particularly helpful for developers looking for
+examples and best practices is `ARIA Patterns
+<https://www.w3.org/WAI/ARIA/apg/patterns/>`__. This web resource contains
+examples of how to implement UI elements—such as menus, dialogs, breadcrumbs,
+and more—in a more accessible way. However, be careful! Just because you can
+implement a button using divs and aria attributes does not mean that you should!
+(Most likely you should just use the button tag.) As a best practice, you should
+only use ARIA when you cannot  use existing HTML elements (button, input, nav,
+aside, etc.) to achieve the UX that you desire.
+
+The rest of this section contains other best practices.
+
+Use color variables from the CSS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When fixing contrast or other visual accessibility issues in JupyterLab, it can
+be tempting to pick a color and apply it to the part of the UI that you are
+working on. However, it quickly becomes unmanageable to have color values spread
+throughout the app across different CSS files. Therefore, the JupyterLab
+codebase defines a set of `color variables
+<https://github.com/jupyterlab/jupyterlab/blob/main/packages/theme-light-extension/style/variables.css>`__
+that can be used for borders, icons, and such. If you're adding any CSS that
+needs a color value, please use one of the variables defined.
+
+Upstream fixes in Lumino
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+JupyterLab does not use React, Vue, or Angular; it uses a front-end framework
+that was built specifically for it called Lumino. So some of the accessibility
+issues tracked in the JupyterLab GitHub repo actually need to be fixed in the
+Lumino repo.
+
+It's not always obvious when an accessibility issue should be fixed in the
+JupyterLab versus Lumino codebase. Generally speaking, if you can fix the issue
+in Lumino, it's better to fix it in Lumino because then the fix will be absorbed
+in more places. However, for that same reason, because Lumino is used by more
+codebases than just JupyterLab—specifically, by JupyterLab extensions—one should
+be careful making changes to Lumino that might break downstream
```

Comment:
I wonder if this is a matter of tone or framing. 

I mean, yes, something is being asked of a developer reading this paragraph: it's to think carefully about code changes to Lumino. 

I didn't want to define breaking change, because it can be a bit fuzzy. A change to the API would definitely be a breaking change, but there are other changes that might not be formally declared as API but if you change it, then you will probably break code that depends on Lumino. For example, Lumino often adds a CSS class to elements when they are hovered to indicate that they are active. Downstream code relies on the presence of that class name to effectively create hover styles. If you change the code so that it stops applying that class, you will break somebody's downstream code.

Reply:
Typical CSS in downstream applications:

```css
.something.lm-mod-active {
  /* change background color */
}
```

Reply:
This is the only unresolved comment. I will tackle this again later, but if you have any thoughts in the meantime, please share them :)